test/e2e/apimachinery/crd_publish_openapi.go

```diff
@@ -833,3 +903,220 @@ properties:
         type: array
         items:
           type: object`)
+
+// setupAPIServersProxyPodAndPortForward a convenience method that creates and runs a pod that proxies connections to the API servers.
+// It also uses kubectl port-forward to route local connections to that pod.
+func setupAPIServersProxyPodAndPortForward(f *framework.Framework) ([]int, func(), error) {
```

Comment:
it's unclear to me why running this code inside a pod would improve stability of this test... something running in a pod can have the same issue reaching an arbitrary API server in an HA environment, and this seems to introduce additional ways the test can fail (issues with the pod, with the kubectl port-forward, etc)

Reply:
it will improve stability because the pod exposes a proxy to all instances (the list is taken from the default endpoint resource) and we check the spec from every replica before running the actual tests.

before we checked a pubic endpoint that was behind a LB